nitro-protocol/contracts/interfaces/IForceMove.sol

```diff
@@ -35,78 +35,64 @@ interface IForceMove {
-     * @param isFinalCount Describes how many of the submitted states have the `isFinal` property set to `true`. It is implied that the rightmost `isFinalCount` states are final, and the rest are not final.
      * @param sigs An array of signatures that support the state with the `largestTurnNum`: one for each participant, in participant order (e.g. [sig of participant[0], sig of participant[1], ...]).
      * @param whoSignedWhat An array denoting which participant has signed which state: `participant[i]` signed the state with index `whoSignedWhat[i]`.
      */
     function checkpoint(
-        FixedPart calldata fixedPart,
-        uint48 largestTurnNum,
-        IForceMoveApp.VariablePart[] calldata variableParts,
-        uint8 isFinalCount, // how many of the states are final
-        Signature[] calldata sigs,
-        uint8[] calldata whoSignedWhat
+        FixedPart memory fixedPart,
+        IForceMoveApp.VariablePart[] memory variableParts,
+        Signature[] memory sigs,
+        uint8[] memory whoSignedWhat
     ) external;
 
     /**
-     * @notice Finalizes a channel by providing a finalization proof.
-     * @dev Finalizes a channel by providing a finalization proof.
-     * @param largestTurnNum The largest turn number of the submitted states; will overwrite the stored value of `turnNumRecord`.
+     * @notice Finalizes a channel by providing a finalization proof. External wrapper for _conclude.
+     * @dev Finalizes a channel by providing a finalization proof. External wrapper for _conclude.
      * @param fixedPart Data describing properties of the state channel that do not change with state updates.
-     * @param appData Application specific data. Applies to all states in the finalization proof.
-     * @param outcome An outcome structure bytes.
+     * @param latestVariablePart Latest variable part in finalization proof. Must have the largest turnNum and the same appData and outcome as all other variable parts in finalization proof.
      * @param numStates The number of states in the finalization proof.
      * @param whoSignedWhat An array denoting which participant has signed which state: `participant[i]` signed the state with index `whoSignedWhat[i]`.
-     * @param sigs An array of signatures that support the state with the `largestTurnNum`:: one for each participant, in participant order (e.g. [sig of participant[0], sig of participant[1], ...]).
+     * @param sigs An array of signatures that support the state with the `largestTurnNum`: one for each participant, in participant order (e.g. [sig of participant[0], sig of participant[1], ...]).
      */
     function conclude(
-        uint48 largestTurnNum,
-        FixedPart calldata fixedPart,
-        bytes memory appData,
-        bytes memory outcome,
+        FixedPart memory fixedPart,
+        IForceMoveApp.VariablePart memory latestVariablePart,
         uint8 numStates,
-        uint8[] calldata whoSignedWhat,
-        Signature[] calldata sigs
```

Comment:
Sure

Reply:
Adding `calldata` to `FixedPart` or `IForceMoveApp.VariablePart` and compilation results in `UnimplementedFeatureError:`. This is pretty strange, because `calldata` is present in current implementation of on-chain statechannels part in monorepo and compiling it finishes successfully.

Reply:
I can't remember, possibly! Happy to just go with `memory` for now if it unblocks us. We can make an issue to circle back to it.